src/OpenLoco/Entities/EntityManager.cpp

```diff
@@ -13,13 +15,73 @@ namespace OpenLoco::EntityManager
     loco_global<uint16_t[numEntityLists], 0x00525E4C> _listCounts;
     loco_global<Entity[maxEntities], 0x006DB6DC> _entities;
     loco_global<EntityId_t[0x40001], 0x01025A8C> _entitySpatialIndex;
+    loco_global<uint32_t, 0x01025A88> _entitySpatialCount;
     static loco_global<string_id, 0x009C68E6> gGameCommandErrorText;
     constexpr size_t _entitySpatialIndexNull = 0x40000;
 
     // 0x0046FDFD
     void reset()
     {
-        call(0x0046FDFD);
+        // Reset all entities to 0
+        std::fill_n(_entities.get(), maxEntities, Entity{});
+        // Reset all entity lists
+        for (auto& count : _listCounts)
+        {
+            count = 0;
+        }
+        for (auto& head : _heads)
+        {
+            head = EntityId::null;
+        }
+
+        // Remake null entities (size maxNormalEntities)
+        EntityBase* previous = nullptr;
+        EntityId_t id = 0;
+        for (; id < maxNormalEntities; ++id)
+        {
+            auto& ent = _entities[id];
+            ent.base_type = EntityBaseType::null;
+            ent.id = id;
+            ent.next_thing_id = EntityId::null;
+            ent.linkedListOffset = static_cast<uint8_t>(EntityListType::null) * 2;
```

Comment:
I don't understand this. The linked list offset is the value of the type, doubled?

Reply:
Its some micro optimisation by CS. Instead of storing the enum value directly he multiplied it by 2 as its always used to access arrays of words which would require it being multiplied by 2. This is exactly the same as RCT2 worked. Thankfully in OpenRCT2 we managed to eventually get rid of this madness.